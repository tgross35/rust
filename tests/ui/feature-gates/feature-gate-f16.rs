#![allow(unused)]

// ensure gating for primitive use
mod m1 {
    const A: f16 = 10.0; //~ ERROR the feature `f16` is unstable

    pub fn main() {
        let a: f16 = 100.0; //~ ERROR the feature `f16` is unstable
        let b = 0.0f16; //~ ERROR the feature `f16` is unstable
        foo(1.23);
    }

    fn foo(a: f16) {} //~ ERROR the feature `f16` is unstable

    struct Bar {
        a: f16, //~ ERROR the feature `f16` is unstable
    }
}

// ensure we don't restrict name as an identifier or custom types
mod m2 {
    #[allow(non_camel_case_types)]
    struct f16 {
        a: i32,
    }

    fn foo() {
        let f16 = ();
    }

    fn bar(a: f16) -> i32 {
        a.a
    }
}

fn main() {}
