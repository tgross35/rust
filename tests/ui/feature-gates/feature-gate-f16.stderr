error[E0658]: the feature `f16` is unstable
  --> $DIR/feature-gate-f16.rs:5:14
   |
LL |     const A: f16 = 10.0;
   |              ^^^
   |
   = note: see issue #116909 <https://github.com/rust-lang/rust/issues/116909> for more information
   = help: add `#![feature(f16)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: the feature `f16` is unstable
  --> $DIR/feature-gate-f16.rs:8:16
   |
LL |         let a: f16 = 100.0;
   |                ^^^
   |
   = note: see issue #116909 <https://github.com/rust-lang/rust/issues/116909> for more information
   = help: add `#![feature(f16)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: the feature `f16` is unstable
  --> $DIR/feature-gate-f16.rs:13:15
   |
LL |     fn foo(a: f16) {}
   |               ^^^
   |
   = note: see issue #116909 <https://github.com/rust-lang/rust/issues/116909> for more information
   = help: add `#![feature(f16)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: the feature `f16` is unstable
  --> $DIR/feature-gate-f16.rs:16:12
   |
LL |         a: f16,
   |            ^^^
   |
   = note: see issue #116909 <https://github.com/rust-lang/rust/issues/116909> for more information
   = help: add `#![feature(f16)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error[E0658]: the feature `f16` is unstable
  --> $DIR/feature-gate-f16.rs:9:17
   |
LL |         let b = 0.0f16;
   |                 ^^^^^^
   |
   = note: see issue #116909 <https://github.com/rust-lang/rust/issues/116909> for more information
   = help: add `#![feature(f16)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0658`.
